server/channels/app/post_file_change.go

```diff
@@ -9,14 +9,21 @@ import (
 	"github.com/mattermost/mattermost/server/public/utils"
 )
 
-func (a *App) processPostFileChanges(rctx request.CTX, newPost, oldPost *model.Post) (model.StringArray, *model.AppError) {
+func (a *App) processPostFileChanges(rctx request.CTX, newPost, oldPost *model.Post, updatePostOptions *model.UpdatePostOptions) (model.StringArray, *model.AppError) {
 	newFileIDs := model.RemoveDuplicateStrings(newPost.FileIds)
 	oldFileIDs := model.RemoveDuplicateStrings(oldPost.FileIds)
 
 	addedFileIDs, removedFileIDs, unchangedFileIDs := utils.FindExclusives(newFileIDs, oldFileIDs)
 
 	if len(addedFileIDs) > 0 {
-		a.attachNewFilesToPost(rctx, newPost, addedFileIDs, unchangedFileIDs)
+		if updatePostOptions.IsRestorePost {
```

Comment:
Fair point, I've added a logic at the root of `UpdatePost` and `PatchPost` functions in App layer to set the param to a new empty struct if its null. This will prevent nil pointer dereference errors and maintain the expected defaults.